test/fixtures/simple-contexts.d.ts

```diff
@@ -0,0 +1,19 @@
+import {Component, PanelLifecycleContext} from '../../lib';
+
+export interface Theme {
+  getName(): string;
+}
+
+export class LightTheme implements Theme {
+  getName(): 'white';
```

Comment:
It would be interesting to see a fuller end-to-end test by using these names in component templates and then checking the rendered content. What do you think?

Reply:
Makes sense. I rendered the theme name into the class name of widget root divs, then augmented existing tests to check the element class name in addition to the direct `getContext()` checks.